(node:21) ExperimentalWarning: CommonJS module /home/container/tmp/node-v23.1.0-linux-x64/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /home/container/tmp/node-v23.1.0-linux-x64/lib/node_modules/npm/node_modules/supports-color/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

up to date, audited 167 packages in 1s

27 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> discord-bot@1.0.0 start
> node index.js

ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± activedevbadge
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± admin-data
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± create_ticket_channels
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± create-welcome
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± force-checkout
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± force-close-tickets
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± help
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± leaderboard
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± open-sessions
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± ping
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± points
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± reset-tickets
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± servers-list
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± setup-apply
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± setup_attendance
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± stats
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± update
ØªÙ… ØªØ­Ù…ÙŠÙ„ Ø§Ù„Ø£Ù…Ø± vac
ØªÙ… ØªØ³Ø¬ÙŠÙ„ Ø§Ù„Ø¯Ø®ÙˆÙ„ Ø¨Ù†Ø¬Ø§Ø­
ØªÙ… Ø¥Ø¹Ø¯Ø§Ø¯ Ø§Ù„Ø¨ÙˆØª Ø¨Ù†Ø¬Ø§Ø­
[32minfo[39m: ØªÙ… ØªØ³Ø¬ÙŠÙ„ Ø§Ù„Ø¯Ø®ÙˆÙ„ ÙƒÙ€ ArabPast Bot#3173 {"timestamp":"2024-11-20T18:59:17.400Z"}
ØªÙ… ØªØ³Ø¬ÙŠÙ„ Ø§Ù„Ø¯Ø®ÙˆÙ„ ÙƒÙ€ ArabPast Bot#3173
[32minfo[39m: ØªÙ… Ø§Ù„Ø§ØªØµØ§Ù„ Ø¨Ù‚Ø§Ø¹Ø¯Ø© Ø§Ù„Ø¨ÙŠØ§Ù†Ø§Øª MongoDB {"timestamp":"2024-11-20T18:59:17.488Z"}
Ø¨Ø¯Ø¡ ØªØ­Ø¯ÙŠØ« 18 Ù…Ù† Ø§Ù„Ø£ÙˆØ§Ù…Ø±.
âœ… ØªÙ… ØªØ­Ø¯ÙŠØ« 18 Ù…Ù† Ø§Ù„Ø£ÙˆØ§Ù…Ø± Ø¨Ù†Ø¬Ø§Ø­.
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-20T19:04:12.159Z"}
Starting check-in process for user: 743432232529559684
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-20T19:04:33.556Z"}
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-20T19:09:15.284Z"}
[32minfo[39m: Starting automatic check-out... {"timestamp":"2024-11-20T20:58:00.242Z"}
[32minfo[39m: Starting daily attendance report... {"timestamp":"2024-11-20T20:59:00.280Z"}
[32minfo[39m: Generating daily statistics... {"timestamp":"2024-11-21T00:00:00.051Z"}
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-21T00:35:28.616Z"}
Starting check-in process for user: 1224036535813279852
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-21T00:42:00.439Z"}
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-21T01:04:22.704Z"}
Starting check-in process for user: 1239339505740419174
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-21T01:04:40.597Z"}
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-21T01:04:42.239Z"}
Starting check-in process for user: 855169577473933322
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-21T01:05:58.581Z"}
Starting check-in process for user: 1239339505740419174
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-21T01:12:07.474Z"}
[32minfo[39m: Button pressed: check_in {"timestamp":"2024-11-21T02:57:13.186Z"}
Starting check-in process for user: 855169577473933322
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-21T02:57:16.380Z"}
[32minfo[39m: Button pressed: check_out {"timestamp":"2024-11-21T09:26:29.103Z"}
[31merror[39m: Uncaught Exception: Uncaught Exception: {"critical":true,"stack":"TypeError: Cannot read properties of undefined (reading 'startsWith')\n    at Client.<anonymous> (/home/container/index.js:2211:37)\n    at Client.emit (node:events:519:35)\n    at InteractionCreateAction.handle (/home/container/node_modules/discord.js/src/client/actions/InteractionCreate.js:97:12)\n    at module.exports [as INTERACTION_CREATE] (/home/container/node_modules/discord.js/src/client/websocket/handlers/INTERACTION_CREATE.js:4:36)\n    at WebSocketManager.handlePacket (/home/container/node_modules/discord.js/src/client/websocket/WebSocketManager.js:348:31)\n    at WebSocketManager.<anonymous> (/home/container/node_modules/discord.js/src/client/websocket/WebSocketManager.js:232:12)\n    at WebSocketManager.emit (/home/container/node_modules/@vladfrangu/async_event_emitter/dist/index.cjs:287:31)\n    at WebSocketShard.<anonymous> (/home/container/node_modules/@discordjs/ws/dist/index.js:1190:51)\n    at WebSocketShard.emit (/home/container/node_modules/@vladfrangu/async_event_emitter/dist/index.cjs:287:31)\n    at WebSocketShard.onMessage (/home/container/node_modules/@discordjs/ws/dist/index.js:1005:14)","timestamp":"2024-11-21T11:12:54.825Z"}
[31merror[39m: Error in interaction: Error in interaction: {"critical":false,"stack":"DiscordAPIError[10062]: Unknown interaction\n    at handleErrors (/home/container/node_modules/@discordjs/rest/dist/index.js:727:13)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async BurstHandler.runRequest (/home/container/node_modules/@discordjs/rest/dist/index.js:831:23)\n    at async _REST.request (/home/container/node_modules/@discordjs/rest/dist/index.js:1272:22)\n    at async ChatInputCommandInteraction.deferReply (/home/container/node_modules/discord.js/src/structures/interfaces/InteractionResponses.js:72:5)\n    at async Object.execute (/home/container/commands/points.js:9:9)\n    at async Client.<anonymous> (/home/container/index.js:300:9)","timestamp":"2024-11-21T11:12:55.048Z"}
Error in interaction: DiscordAPIError[10062]: Unknown interaction
    at handleErrors (/home/container/node_modules/@discordjs/rest/dist/index.js:727:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async BurstHandler.runRequest (/home/container/node_modules/@discordjs/rest/dist/index.js:831:23)
    at async _REST.request (/home/container/node_modules/@discordjs/rest/dist/index.js:1272:22)
    at async ChatInputCommandInteraction.deferReply (/home/container/node_modules/discord.js/src/structures/interfaces/InteractionResponses.js:72:5)
    at async Object.execute (/home/container/commands/points.js:9:9)
    at async Client.<anonymous> (/home/container/index.js:300:9) {
  requestBody: { files: undefined, json: { type: 5, data: [Object] } },
  rawError: { message: 'Unknown interaction', code: 10062 },
  code: 10062,
  status: 404,
  method: 'POST',
  url: 'https://discord.com/api/v10/interactions/1309114314296266812/aW50ZXJhY3Rpb246MTMwOTExNDMxNDI5NjI2NjgxMjo1bXJSQnlUWnVDYXJTVTBzT0VWT0lDWmhjSlEzQmlJeUhvUDlPdlVUM1d2VlIycGoydnNnb2xOMDNNcTJmZ2dDZzdSZGlTMlJpWDhycEJLQU5INjd1bmlTYUlJMm5QR2p4eUd4T3p3QTFqZ1FHNWJaTG5rYTh3am5aSHh1ckFiVg/callback'
}
Error in error handler: DiscordAPIError[40060]: Interaction has already been acknowledged.
    at handleErrors (/home/container/node_modules/@discordjs/rest/dist/index.js:727:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async BurstHandler.runRequest (/home/container/node_modules/@discordjs/rest/dist/index.js:831:23)
    at async _REST.request (/home/container/node_modules/@discordjs/rest/dist/index.js:1272:22)
    at async ChatInputCommandInteraction.reply (/home/container/node_modules/discord.js/src/structures/interfaces/InteractionResponses.js:115:5)
    at async handleInteractionError (/home/container/index.js:1189:13)
    at async Client.<anonymous> (/home/container/index.js:303:9) {
  requestBody: { files: [], json: { type: 4, data: [Object] } },
  rawError: {
    message: 'Interaction has already been acknowledged.',
    code: 40060
  },
  code: 40060,
  status: 400,
  method: 'POST',
  url: 'https://discord.com/api/v10/interactions/1309114314296266812/aW50ZXJhY3Rpb246MTMwOTExNDMxNDI5NjI2NjgxMjo1bXJSQnlUWnVDYXJTVTBzT0VWT0lDWmhjSlEzQmlJeUhvUDlPdlVUM1d2VlIycGoydnNnb2xOMDNNcTJmZ2dDZzdSZGlTMlJpWDhycEJLQU5INjd1bmlTYUlJMm5QR2p4eUd4T3p3QTFqZ1FHNWJaTG5rYTh3am5aSHh1ckFiVg/callback'
}
